### ISSUE TITLE
B905 not being raised for same iterable in `zip`

### ISSUE THREAD

**Issue by jamesbraza** [charliermarsh/ruff#4102]
With `flake8==6.0.0`, `flake8-bugbear==23.3.23`, `ruff==0.0.263`:

```bash
> flake8 a.py
> ruff --select=B905 a.py
a.py:2:21: B905 `zip()` without an explicit `strict=` parameter
Found 1 error.
```

```python
some_list = [1, 2, 3]
for first_second in zip(some_list, some_list[1:]):
    print(first_second)
```

I know the right move overall is `itertools.pairwise`.

However, in isolation, I think `B905` should not be triggered here.  It should know that the default `strict=False` is the right move here.

Aside: `ruff` triggers here while `flake8` does not.  Not sure why.

**Comment by evanrittenhouse** [comment]
I can look into this. If there's a divergence between ruff and flake8, I'd consider it a bug

**Comment by evanrittenhouse** [comment]
Isn't B905 complaining about *explicit* stricts though? Just thinking that if the rule expects an explicit strict, it probably should trigger regardless of the keyword argument's default value.

**Comment by charliermarsh** [comment]
Hmm, yeah, I think the current behavior is reasonable. Note that `flake8-bugbear` _does_ behave the same way, but you have to explicitly enable it, since it's an error code that starts with `9` (as in, `B905`), a behavior that Ruff doesn't match:

```shell
❯ flake8 foo.py --select B905
foo.py:2:21: B905 `zip()` without an explicit `strict=` parameter
```

**Comment by jamesbraza** [comment]
Oh dang, didn't know that `flake8-bugbear` has default-disabled codes.  Sorry for that.

I guess I made this issue because, if one listens to the message and without paying attention and marks `strict=True`, it will result in a `ValueError`at runtime:

```python
some_list = [1, 2, 3]
for first_second in zip(some_list, some_list[1:], strict=True):
    pass

# Leads to ValueError: zip() argument 2 is shorter than argument 1
```

To ameliorate this, it requires making a special case to only suggest `strict=False`, so I can see arguments for just leaving this false-positive-ish message around.

Of course, the right action is to use `itertools.pairwise`.  Just posting for posterity, cheers!

**Comment by jaxwagner** [comment]
Hi! I'm having an issue where rule B905 is not raising where I think it should be. I can get B905 to flag in the playground, but not via my ruff.toml config file. 

here's an example where I think it should flag and it isnt with my config:
`pairs = zip(self._points[:-1], self._points[1:])`

and here is my config, where B905 is not disabled:
```
[lint]
extend-select = ["I", "W", "B"]
ignore = ["B904", "B019", "B008", "B015", "B007", "B011", "B012", "B010", "B006", "B024"]
```
is there something else I need to do to enable it? many other bugbear rules are working as expected, so I'm not sure why B905 is not. thanks so much for any help!

**Comment by charliermarsh** [comment]
You may need to set your minimum-supported Python version to Python 3.10. Do you have something like this in your config?

```toml
target-version = "py310"
```

Alternatively, if you set a `requires-python` in your `pyproject.toml`, Ruff will respect that too.

**Comment by jaxwagner** [comment]
thanks so much @charliermarsh! really appreciate it :)
